crates/gpui/src/platform/linux/wayland/window.rs

```diff
@@ -308,8 +314,77 @@ impl PlatformWindow for WaylandWindow {
         msg: &str,
         detail: Option<&str>,
         answers: &[&str],
-    ) -> Receiver<usize> {
-        unimplemented!()
+    ) -> oneshot::Receiver<usize> {
+        let (done_tx, done_rx) = oneshot::channel();
+        let msg = msg.to_string();
+        let detail = detail.unwrap_or_default().to_string();
+        let mut ans: Vec<String> = Vec::new();
+        let mut ans_idx: Vec<usize> = Vec::new();
```

Comment:
This may look a bit weird. I am taking the value from reference(`&`) arguments and storing it, because when I try to use them directly inside my `let dialog = move || unsafe {...}` block below, I get lifetime issues saying "`msg` is a reference that is only valid in the method body". So, I convert the pointer to `String` or `Vec<String>` and then use it.